source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -1242,6 +1243,31 @@ message TransformationReplaceLinearAlgebraInstruction {
 
 }
 
+message TransformationReplaceParamsWithStruct {
+
+  // Replaces parameters of the function with a struct containing
+  // values of those parameters.
+
+  // Result ids of parameters to replace.
+  repeated uint32 parameter_id = 1;
+
+  // Fresh id for a new function type. This might be unused if the required type
+  // already exists in the module or if we can change the old type.
+  uint32 fresh_function_type_id = 2;
+
+  // Fresh id for a new struct function parameter to be used as a replacement.
+  uint32 fresh_parameter_id = 3;
+
+  // Fresh ids for struct objects containing values of replaced parameters.
+  // This map contains a fresh id for at least every result id of a relevant
+  // OpFunctionCall instruction.
+  //
+  // Current implementation of the transformation does not have an non-deterministic
+  // behaviour. Change to repeated UInt32Pair if the behaviour changes.
```

Comment:
I would change to say: "While maps are not fully deterministic, the way this map is used does not exhibit nondeterminism.  Change to repeated Uint32Pair if this changes."

Reply:
Fixed in 97a33fa.